cosmic-usage-core/src/main/java/com/github/missioncriticalcloud/cosmic/usage/core/services/TokenService.java

```diff
@@ -0,0 +1,17 @@
+package com.github.missioncriticalcloud.cosmic.usage.core.services;
+
+import java.io.IOException;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.github.missioncriticalcloud.cosmic.usage.core.model.DomainToken;
+
+public interface TokenService {
+
+    String encryptDomainToken(final String domain) throws JsonProcessingException;
```

Comment:
I'd rename this method to just `String encrypt(DomanToken)` and I'd wrap vendor-specific exceptions like `JsonProcessingException` with something like `TokenException`